system-int-test/test/cmr/system_int_test/search/granule_retrieval_test.clj

```diff
@@ -35,17 +45,85 @@
-                                      (search/retrieve-concept
-                                        "G1111-PROV1" nil {:throw-exceptions true}))]
+                                     (search/retrieve-concept
+                                      "G1111-PROV1" nil {:throw-exceptions true}))]
         (is (= 404 status))
-        (is (= ["Concept with concept-id [G1111-PROV1] could not be found."] errors))))))
+        (is (= ["Concept with concept-id [G1111-PROV1] could not be found."] errors))))
+    (testing "retrieval by un-supported accept header."
+      (let [{:keys [status errors]} (search/get-search-failure-xml-data
+                                     (search/retrieve-concept
+                                      (:concept-id gran1)
+                                      (:revision-id gran1)
+                                      {:accept "application/atom"
+                                       :throw-exceptions true}))]
+        (is (= 400 status))
+        (is (= ["The mime types specified in the accept header [application/atom] are not supported."]
+               errors))))))
+
+(deftest retrieve-umm-g-granule-test
+  (let [collection (d/ingest-umm-spec-collection
+                    "PROV1" (data-umm-c/collection {:EntryTitle "correct"
+                                                    :ShortName "S1"
+                                                    :Version "V1"}))
+        granule (-> (dg/granule-with-umm-spec-collection
+                     collection
+                     (:concept-id collection)
+                     {:granule-ur "Gran1"
+                      :collection-ref (umm-g/map->CollectionRef {:entry-title "correct"})}))
+        echo10-gran (d/item->concept granule :echo10)
+        umm-g-gran (d/item->concept granule :umm-json)
+        {echo10-gran-concept-id :concept-id
+         echo10-gran-revision-id :revision-id} (ingest/ingest-concept echo10-gran)
+        {umm-g-gran-concept-id :concept-id
+         umm-g-gran-revision-id :revision-id} (ingest/ingest-concept umm-g-gran)]
+    (index/wait-until-indexed)
+    (testing "retrieve UMM-G granule in UMM JSON format"
+      (are3
+        [accept-format]
+        (let [response (search/retrieve-concept
+                        umm-g-gran-concept-id umm-g-gran-revision-id {:accept accept-format})
+              response-format (-> response
+                                  (get-in [:headers :Content-Type])
+                                  mt/mime-type->format)]
+          (is (= {:format :umm-json :version "1.4"}
+                 response-format))
+          (is (= (:metadata umm-g-gran) (:body response))))
+
+        "without specifying accept format"
+        nil
+
+        "specifying umm json accept format without version"
+        "application/vnd.nasa.cmr.umm+json"
+
+        "specifying umm json accept format with version"
+        "application/vnd.nasa.cmr.umm+json;version=1.4"))
```

Comment:
Do we already have tests that verify all potential versions can be returned successfully? It seems like that would be worthwhile.

Reply:
There is only one version for UMM-G and it is `1.4` oddly.